Sources/Tracking/CustomerIO.swift

```diff
@@ -140,62 +146,70 @@ public class CustomerIO: CustomerIOInstance {
      Initialize the shared `instance` of `CustomerIO`.
      Call this function when your app launches, before using `CustomerIO.instance`.
      */
+    @available(iOSApplicationExtension, unavailable)
     public static func initialize(
         siteId: String,
         apiKey: String,
-        region: Region = Region.US
+        region: Region = Region.US,
+        configure configureHandler: ((inout SdkConfig) -> Void)? = nil
     ) {
+        var newSdkConfig = SdkConfig.Factory.create(region: region)
+
+        if let configureHandler = configureHandler {
+            configureHandler(&newSdkConfig)
+        }
+
         Self.initialize(
             siteId: siteId,
             apiKey: apiKey,
             region: region,
-            configureHandler: nil
+            isFromiOSApplicationExtension: false,
+            config: newSdkConfig
         )
     }
 
     /**
      Initialize the shared `instance` of `CustomerIO`.
-     Call this function when your app launches, before using `CustomerIO.instance`.
+     Call this function in your Notification Service Extension for the rich push feature.
      */
-    @available(iOSApplicationExtension, unavailable)
-    public static func initialize(
+    public static func initializeRichPush(
```

Comment:
Added an `initialize` that is dedicated for a Notification Service Extension. I tried to create 2 versions of `initialize()` instead of creating a separately named `initializeRichPush` but had no luck. The compiler wouldn't compile the SDK code. 

I tried to create 1 initialize that is only available to NSE and another only available to host iOS apps. However, because the method signature is identical for each target: 

```swift
// For host app
initialize(siteid: String, apiKey: String, region?: Region, config?)

// For NSE
initialize(siteid: String, apiKey: String, region?: Region, config?)
```

the compiler wouldn't allow this type of code. 

Another solution that we could have used is to remove some optional parameters to make the compiler notice there are 2 separate signatures:

```swift
// For host app
initialize(siteid: String, apiKey: String, region: Region, config: (SDKConfig) -> Void)

// For NSE
initialize(siteid: String, apiKey: String, region: Region, config: (RichPushSDKConfig) -> Void)
```

But I wasn't sure that we would like this solution, either. We would still be introducing a breaking change to customers.

Reply:
> We use [our API to find your region](https://customer.io/docs/api/#operation/getRegion) for your workspace so we don't need to ask for it anymore.

Not sure if we would like to do this additional step ourselves but I found 👇  useful if one is available only in host target and the other one only in NSE

```
// For host app
initialize(siteid: String, apiKey: String, region: Region, config: (SDKConfig) -> Void)

// For NSE
initialize(siteid: String, apiKey: String, region: Region, config: (RichPushSDKConfig) -> Void) 
```

Reply:
I believe what you're referring to is "Another solution that we could have used is to remove some optional parameters to make the compiler notice there are 2 separate signatures:" from [my original post](https://github.com/customerio/customerio-ios/pull/220#discussion_r1029791315)?

Reply:
Yes, correct ! I found that one more convenient to use from a dev's perspective.

Reply:
Option 2 has been implemented.